modules/phase_field/src/userobjects/EBSDReader.C

```diff
@@ -255,7 +255,7 @@ EBSDReader::getGrainNum() const
 }
 
 unsigned int
-EBSDReader::getGrainNum(unsigned int phase) const
+EBSDReader::getGrainNumPhase(unsigned int phase) const
```

Comment:
This was discussed a couple week ago.  See the conversation at:
https://github.com/idaholab/moose/pull/12494#issuecomment-454856863
https://github.com/idaholab/moose/pull/12494#issuecomment-454872775

Reply:
Ok, going back over teh code I see the issue. Making `unsigned int getGrainNum(unsigned int phase) const;` virtual will trigger the warning because a virtual function with this signature does not exist in `EulerAngleProvider`. And since I was not excited about adding it there you needed to rename. Unfortunately I don't like either way (reinforcing my feeling that the entire design is rotten), but I guess I'll have to get over it...

Reply:
I don't like the design either but am trying to work within the EulerAngleProvider base class.  The other option would be to create a derivative version of EulerAngleProvider2RGBAux for EBSD datasets that bypasses EulerAngleProvider and directly calls EBSDReader.

Reply:
A second option would be to make the changes in FeatureFloodCount.C so that the feature_id has the correct index before it is output.

Reply:
Let's not touch FFC. One thing I suggested a while ago (to Cody maybe) was to do a dynamic cast from `EulerAngleProvider` to `EBSDReader` if the phase parameter is supplied (with error checking). Then you can keep it all in one class, don't have to touch the method name (and keep it non-virtual)

Reply:
I added the dynamic cast as you suggested.  Please see my latest PR.